packages/mui-material/src/Button/buttonClasses.ts

```diff
@@ -176,6 +176,20 @@ export interface ButtonClasses {
   colorInfo: string;
   /** Styles applied to the root element if `color="warning"`. */
   colorWarning: string;
+  /** Styles applied to the root element if `loading={true}`. */
+  loading: string;
+  /** Styles applied to the loadingWrapper element. */
+  loadingWrapper: string;
+  /** Styles applied to the loadingIconPlaceholder element. */
+  loadingIconPlaceholder: string;
+  /** Styles applied to the loadingIndicator element. */
+  loadingIndicator: string;
+  /** Styles applied to the loadingIndicator element if `loadingPosition="center"`. */
+  loadingIndicatorCenter: string;
+  /** Styles applied to the loadingIndicator element if `loadingPosition="start"`. */
+  loadingIndicatorStart: string;
+  /** Styles applied to the loadingIndicator element if `loadingPosition="end"`. */
+  loadingIndicatorEnd: string;
```

Comment:
These should be removed and replaced by

`loadingPositionCenter`, `loadingPositionStart`, `loadingPositionEnd`, applied to the root. This would allow to do

```
.loadingPositionCenter .loadingIndicator { ... } // instead of loadingIndicatorCenter

// but also
.loadingPositionCenter .loadingWrapper { ... }

// or
.loadingPositionCenter .icon { ... }
```

This follows the prop-value class combination.

Reply:
Good point, fixed.

Reply:
@siriwatknp the `loadingPositionCenter`, `loadingPositionStart`, and `loadingPositionEnd` classes should be **applied to the root**. Otherwise, it won't be possible to compose it with the `icon`, `loadingWrapper`, or `loadingIconPlaceholder` classes.